src/main.rs

```diff
@@ -1,81 +1,113 @@
 use clap::Parser;
+use indexmap::IndexMap;
 use log::info;
 use url::Url;
 
-use springbok_mgl::{create_refiner_map, print_entries_or_append_query_pair, Cli, get_and_print_search_results};
+use springbok_mgl::{
+    create_refiner_map, download_bill_text, get_and_print_search_results,
+    print_entries_or_append_query_pair, Cli,
+};
 
 fn main() {
     env_logger::init();
 
     // Get default search page, parse, and create refiner map
-
     info!("Creating refiner map");
     let refiner_map = create_refiner_map();
 
-    // Parsing command line arguments and construct search URL
-
+    // Parse command line arguments and construct search URL
     info!("Constructing search URL");
     let cli = Cli::parse();
     let mut search_url = Url::parse("https://malegislature.gov/Bills/Search").unwrap();
 
     // https://malegislature.gov/Bills/Search?SearchTerms=&Page=1
     // https://malegislature.gov/Bills/Search?SearchTerms=mbta&Page=1
-    match cli.search_term {
-        None => search_url.set_query(Some("SearchTerms=&Page=1")),
+    let mut do_search;
```

Comment:
This mut do_search feels a little odd to me. Would it be better to call print_entries_or_append_query_pair where the searching and printing/appending wasn't a side effect?

Reply:
I.e., return an option, and then continue searching and append etc. based on a match with that option. I could be misunderstanding what's happening in the code though.